src/mongo/db/ldap/ldap_manager_impl.cpp

```diff
@@ -551,15 +553,19 @@ Status LDAPManagerImpl::execQuery(std::string& ldapurl, std::vector<std::string>
     LDAPMessage*answer = nullptr;
     LDAPURLDesc *ludp{nullptr};
     int res = ldap_url_parse(ldapurl.c_str(), &ludp);
-    ON_BLOCK_EXIT([&] { ldap_free_urldesc(ludp); return_search_connection(ldap); });
+    // 'ldap' should be captured by reference
```

Comment:
This is because `ldap` is "returned" and reassigned on lines 591 and 592, isn't it? If so, I'd mention that in the comment.